src/Symfony/Component/DomCrawler/Form.php

```diff
@@ -197,9 +197,21 @@ public function getUri()
     {
         $uri = parent::getUri();
 
-        if (!in_array($this->getMethod(), array('POST', 'PUT', 'DELETE', 'PATCH')) && $queryString = http_build_query($this->getValues(), null, '&')) {
-            $sep = false === strpos($uri, '?') ? '?' : '&';
-            $uri .= $sep.$queryString;
+        if (!in_array($this->getMethod(), array('POST', 'PUT', 'DELETE', 'PATCH'))) {
+            $query = parse_url($uri, PHP_URL_QUERY);
+            $currentParameters = array();
+            if ($query) {
+                foreach (explode('&', $query) as $param) {
```

Comment:
But if you use the second optional argument of the function, the variables are stored in an array, right?